templates/common/openstack/files/usr-local-bin-openstack-kubelet-nodename.yaml

```
@@ -31,16 +31,38 @@ contents:
     # Set node name to be instance name instead of the default FQDN hostname.
     # Try to get the hostname from config drive first if it's available
     # or fallback to the metadata service.
-    if [ -f "/host/var/config/openstack/2012-08-10/meta_data.json" ]; then
+    # We'll first try to use config-drive: if a block device is found with a specific label,
+    # we'll mount it and unmount it when not needed anymore. This is because the config drive is
+    # unmounted by cloud-init.
+    # If config-drive isn't found, then we'll try to reach the metadata server.
+    if [ -n "$(blkid -t LABEL="config-2" -odevice)" ]; then
+        config_drive_label="config-2"
+    elif [ ! -z "$(blkid -t LABEL="CONFIG-2" -odevice)" ]; then
```

Stephen's suggestion could also be applied here. 
```suggestion
    elif [ -n "$(blkid -t LABEL="CONFIG-2" -odevice)" ]; then
```